bindings/java/c/host.c

```diff
@@ -0,0 +1,846 @@
+#include <assert.h>
+#include <stdlib.h>
+
+#include "host.h"
+
+static JNIEnv* attach()
+{
+    JNIEnv* jenv;
+    jint rs = (*jvm)->AttachCurrentThread(jvm, (void**)&jenv, NULL);
+    assert(rs == JNI_OK);
+    return jenv;
+}
+
+static void detach(JNIEnv* jenv)
+{
+    if ((*jenv)->ExceptionOccurred(jenv))
+    {
+        (*jenv)->ExceptionDescribe(jenv);
+    }
+    (*jvm)->DetachCurrentThread(jvm);
+}
+
+static bool account_exists_fn(struct evmc_host_context* context, const evmc_address* address)
+{
+    bool result = false;
+    const char java_method_name[] = "account_exists";
+    const char java_method_signature[] = "(I[B)I";
+    JNIEnv* jenv = attach();
+    if (jenv != NULL)
+    {
+        jclass host_class;
+        jmethodID method;
+        jint jcontext_index;
+        jbyteArray jaddress;
+
+        // get java class
+        host_class = (*jenv)->FindClass(jenv, "org/ethereum/evmc/Host");
+        if (!host_class)
+        {
+            printf("JNI Error: FindClass was unable to load the Host class\n");
+            detach(jenv);
+        }
+
+        // get java method
+        method =
+            (*jenv)->GetStaticMethodID(jenv, host_class, java_method_name, java_method_signature);
+        if (!method)
+        {
+            printf("JNI Error: GetStaticMethodID was unable to find method %s\n", java_method_name);
+            detach(jenv);
```

Comment:
What happens in these cases? Is execution aborted? Or do you still need to exit the function properly?

Reply:
I'm pretty sure it isn't exiting gracefully.  After it raises an exception and detaches, execution will resume and prob cause a segfault.  I will fix this.

Reply:
Can actually any recovery happen at these cases? If not, would a plain `assert(method != NULL)` good as a start? And later we can think about better error recovery once this PR is merged.

My preference would be to take the simplest and quickest approach to always cleanly resolve (even if it is an assert) and we can fine tune this in subsequent PRs later.

Reply:
10-4 will do

Reply:
To clarify, are you suggesting putting the assert() after the detach()?  Or simply replace the entire conditional with the assert() and worry about more fine grained error handling later?

Reply:
ok, I did the latter and pushed to a new commit.

Reply:
Yes, I meant the latter, thanks!